pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -73,957 +71,765 @@
+        )
 
-    def _validate_accelerator_type(self) -> None:
-        if self._accelerator_type and self._accelerator_type != self._device_type:
-            # internal error: should not happen.
-            raise ValueError(
-                f"Mismatch between the requested accelerator type ({self._accelerator_type})"
-                f" and assigned device type ({self._device_type})."
-            )
-        self._accelerator_type = self._device_type
-
-    def _warn_if_devices_flag_ignored(self) -> None:
-        if self.devices is None:
-            return
-        devices_warning = f"The flag `devices={self.devices}` will be ignored, as you have set"
-        if self.distributed_backend in ("auto", _AcceleratorType.TPU):
-            if self.tpu_cores is not None:
-                rank_zero_warn(f"{devices_warning} `tpu_cores={self.tpu_cores}`")
-        elif self.distributed_backend in ("auto", _AcceleratorType.IPU):
-            if self.ipus is not None:
-                rank_zero_warn(f"{devices_warning} `ipus={self.ipus}`")
-        elif self.distributed_backend in ("auto", _AcceleratorType.GPU):
-            if self.gpus is not None:
-                rank_zero_warn(f"{devices_warning} `gpus={self.gpus}`")
-        elif self.distributed_backend in ("auto", _AcceleratorType.CPU):
-            if self.num_processes != 1:
-                rank_zero_warn(f"{devices_warning} `num_processes={self.num_processes}`")
-
-    def _set_devices_if_none(self) -> None:
-        if self.devices is not None:
-            return
-        if self._accelerator_type == _AcceleratorType.TPU:
-            self.devices = self.tpu_cores
-        elif self._accelerator_type == _AcceleratorType.IPU:
-            self.devices = self.ipus
-        elif self._accelerator_type == _AcceleratorType.GPU:
-            self.devices = self.gpus
-        elif self._accelerator_type == _AcceleratorType.CPU:
-            self.devices = self.num_processes
-
-    def _handle_accelerator_and_strategy(self) -> None:
-        deprecated_types = [t for t in _StrategyType if t not in (_StrategyType.TPU_SPAWN, _StrategyType.DDP_CPU)]
-        if self.distributed_backend is not None and self.distributed_backend in deprecated_types:
-            rank_zero_deprecation(
-                f"Passing `Trainer(accelerator={self.distributed_backend!r})` has been deprecated"
-                f" in v1.5 and will be removed in v1.7. Use `Trainer(strategy={self.distributed_backend!r})` instead."
-            )
-            if self._strategy_flag is not None:
+        # --Accelerator-------------------------------------------------------------
+        # handle `auto` and `None`
+        if self._accelerator_flag == "auto" or self._accelerator_flag is None:
+            self._accelerator_flag = self._choose_accelerator()
+        # else:
+        #     # [RFC] move to XAccelerator class init?
+        #     self._check_device_availibility()
```

Comment:
don't forget to open a rfc for that and delete this comment :)

Reply:
Ananth already send out PRs for all of the accelerators #11798 #11797 and etc :)